### PR TITLE
[Backport 2.x] Add support for Node.js >=14.20.1 <19 (#4071)

### PR DESCRIPTION
**Pull request by kavilla**
Manual backport of 1c0ffeeb5b306f6a850f76c77f41be3f81ddd8c2 for https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4071

### CODE REVIEW

**Comment by ps48** [comment]
@kavilla @AMoo-Miki  We see main CI failing already for dashboards plugin with this PR: https://github.com/opensearch-project/dashboards-observability/actions/runs/5092646146/jobs/9158330893?pr=489. Do all plugins need to update their CI with new node version for the 2.8 release?

**Comment by kavilla** [comment]
> @kavilla @AMoo-Miki We see main CI failing already for dashboards plugin with this PR: https://github.com/opensearch-project/dashboards-observability/actions/runs/5092646146/jobs/9158330893?pr=489. Do all plugins need to update their CI with new node version for the 2.8 release?

Correct

**Review by ananzh — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG NODE_VERSION=14.20.1
+ARG NODE_VERSION=18.16.0
```
Comment (AMoo-Miki) [inline Dockerfile:1]: Shouldn’t this be 16.20.0 too?

**Review by AMoo-Miki — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (kavilla) [reply Dockerfile:1]: It should but I think there is a general follow up to pull this correctly. I don't think we have verified if this Dockerfile works

**Pull request merged**